builtin/logical/transit/policy.go

```diff
@@ -21,6 +22,200 @@ const (
 	kdfMode = "hmac-sha256-counter"
 )
 
+// policyCache implements a simple locking cache of policies
+type policyCache struct {
+	cache map[string]*lockingPolicy
+	lock  sync.RWMutex
+}
+
+// loadStoredPolicies loads stored policies into the cache. This should only be
+// run at backend initialization time.
+func (p *policyCache) loadStoredPolicies(storage logical.Storage) error {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+
+	policyNames, err := storage.List("policy/")
```

Comment:
This will miss any policies that contain a slash e.g. the "foo/bar" role, since it won't recurse. Why not just lazy load these in and avoid the problem?